src/custom/pages/Profile/index.tsx

```diff
@@ -223,13 +227,24 @@ export default function Profile() {
             </span>
           </BalanceDisplay>
           <CardActions>
-            <ExtLink href={getBlockExplorerUrl(chainId || 1, COW_CONTRACT_ADDRESS[chainId || 1], 'address')}>
+            <ExtLink
+              title="View contract"
+              href={getBlockExplorerUrl(chainId || 1, COW_CONTRACT_ADDRESS[chainId || 1], 'address')}
+            >
               Contract ↗
             </ExtLink>
 
-            <AddToMetamask currency={COW[chainId || 1] as Currency | undefined} />
+            {library?.provider?.isMetaMask && (
```

Comment:
i think the useWalletInfo hook is better here (where u use `useActiveWeb3React`) as it includes all this data

Reply:
In this case I mimicked logic we already use elsewhere:
`src/custom/components/AddToMetamask/index.tsx` L68

WDYT?

Reply:
all good then 👍